topdown/time.go

```diff
@@ -55,7 +82,12 @@ func builtinTimeParseRFC3339Nanos(a ast.Value) (ast.Value, error) {
 		return nil, err
 	}
 
-	return ast.Number(int64ToJSONNumber(result.UnixNano())), nil
+	ns, err := toSafeUnixNano(result)
+	if err != nil {
+		return nil, err
+	}
+
+	return ast.Number(int64ToJSONNumber(ns)), nil
```

Comment:
This still looks very similar in the three places, but we can't put more logic into the helper function because these two, `builtinTimeParseNanos` and `builtinTimeParseRFC3339Nanos`, are using the (deprecated) `RegisterFunctionalBuiltin*` methods.

What do you think, would be worthwhile to update the builtin code and go with some helper like this?
```go
func toSafeUnixNano(t time.Time, iter func(*ast.Term) error) error {
	// ... checks here

	ns, err := toSafeUnixNano(result)
	if err != nil {
		return err
	}
	return iter(ast.NewTerm(ast.Number(int64ToJSONNumber(ns))))
}
```

Reply:
Sounds good. I'll make that change.